src/transformers/models/bloom/modeling_bloom.py

```diff
@@ -760,6 +770,12 @@ def __init__(self, config):
         # Initialize weights and apply final processing
         self.post_init()
 
+    def tie_weights(self):
+        super(BloomForCausalLM, self).tie_weights()
+
+        if self.config.force_lm_head_in_fp32:
+            self.lm_head.to(torch.float32)
```

Comment:
This will also work with `device_map="auto"` since this happens after weights are loaded, so good for me.